test/tmetrics.m

```diff
@@ -0,0 +1,349 @@
+classdef tmetrics < matlab.unittest.TestCase
+    % tests for traces and spans
+
+    % Copyright 2023 The MathWorks, Inc.
+
+    properties
+        OtelConfigFile
+        OtelRoot
+        JsonFile
+        PidFile
+	OtelcolName
+        Otelcol
+        ListPid
+        ReadPidList
+        ExtractPid
+        Sigint
+        Sigterm
+    end
+
+    methods (TestClassSetup)
+        function setupOnce(testCase)
+            commonSetupOnce(testCase);
+        end
+    end
+
+    methods (TestMethodSetup)
+        function setup(testCase)
+            commonSetup(testCase);
+        end
+    end
+
+    methods (TestMethodTeardown)
+        function teardown(testCase)
+            commonTeardown(testCase);
+        end
+    end
+
+    methods (Test)
+        
+        function testCounterBasic(testCase)
+            % test names and added value in Counter
+
+            metername = "foo";
+            countername = "bar";
+
+            p = opentelemetry.sdk.metrics.MeterProvider();
+            mt = p.getMeter(metername);
+            ct = mt.createCounter(countername);
+
+            % verify MATLAB object properties
+            verifyEqual(testCase, mt.Name, metername);
+            verifyEqual(testCase, mt.Version, "");
+            verifyEqual(testCase, mt.Schema, "");
+            verifyEqual(testCase, ct.Name, countername);
+
+            % create testing value 
+            val = 10;
+
+            % add value and attributes
+            ct.add(val);
+
+            % wait for default collector response time (60s)
+            pause(70);
+
+            % fetch result
+            results = readJsonResults(testCase);
+            results = results{1};
+
+            % verify meter and counter names
+            verifyEqual(testCase, string(results.resourceMetrics.scopeMetrics.metrics.name), countername);
+            verifyEqual(testCase, string(results.resourceMetrics.scopeMetrics.scope.name), metername);
+
+            % fetch datapoint
+            dp = results.resourceMetrics.scopeMetrics.metrics.sum.dataPoints;
+
+            % verify counter value
+            verifyEqual(testCase, dp.asDouble, val);
+
+        end
+
+
+        function testCounterAddAttributes(testCase)
+            % test names, added value and attributes in Counter
+
+            metername = "foo";
+            countername = "bar";
+
+            p = opentelemetry.sdk.metrics.MeterProvider();
+            mt = p.getMeter(metername);
+            ct = mt.createCounter(countername);
+
+            % verify MATLAB object properties
```

Comment:
These tests on lines 93-96 are unnecessary since they are duplicates of lines 50-54.